doc/acl/Extend-L3V6ACLs.md

```diff
@@ -0,0 +1,428 @@
+        - [Cons](#cons-1)
+      - [Option-C: Create a _new_ ACL Table Type *L3V4V6*](#option-c-create-a-new-acl-table-type-l3v4v6)
+        - [Phase 1:](#phase-1)
+        - [Phase 2:](#phase-2)
+        - [Pros](#pros-2)
+      - [Implementation](#implementation)
+        - [Orchagent](#orchagent)
+        - [STATE\_DB](#state_db)
+    - [SAI API](#sai-api)
+    - [Configuration and management](#configuration-and-management)
+      - [YANG model changes](#yang-model-changes)
+    - [CLI](#cli)
+        - [ACL table create CLI](#acl-table-create-cli)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+
+### Revision  
+
+| Rev | Date     | Author          | Change Description |
+|:---:|:--------:|:---------------:|--------------------|
+| 0.1 | 18/Feb/23 | Ravindranath (**Marvell**)   | Initial Version.  |
+| 0.2 | 21/Mar/23 | Ravindranath (**Marvell**)   | Updates based on the community discussion on 21/Feb/23 and other offline comments  |
+
+### Scope  
+
+
+This document discusses the various options to combine L3 and L3V6 ACL tables in SAI on supported hardware. It then provides the high level design in SONiC to support a new ACL table type called **L3V4V6**.
+
+### Terminology
+
+
+
+| SONiC User ACL Table Types     | Description |
+|---------------|--------------------|
+| __L3__ ACL table type | A built-in ACL table type in SONiC that can match __IPv4__ fields along with incoming port and support ACL redirect and packet ACL actions.|
+| __L3V6__ ACL  table type | A built-in ACL table type in SONiC that can match __IPv6__ fields along with incoming port  and support redirect and packet ACL actions.|
+| __Mirror__ ACL  table type | A built-in ACL table type in SONiC that can match __IPv4__ fields and supports __Mirror__ ACL action. |
+| __MirrorV6__ ACL  table type | A built-in ACL table type in SONiC that can match __IPv6__ fields and supports __Mirror__ ACL action. |
+| __Custom__ ACL  table types | A recent enhancement where one or more ACL table types can be created by the user specifying the match-field list, action-list and the bindpoint-types list.|
+
+
+### Overview 
+
+This document describes the orchagent support by which user can create a  new ACL table of type L3V4V6 on platforms that support this feature.
+
+In several ASICs,  IPv4 and IPv6 ACL rules can be supported in the same ACL table. Further, in many hardware, when the hardware tables (TCAMs) are configured to match IPv6 addresses, the hardware can use the same resources to match the corresponding IPv4 packet fields without incurring additional hardware resources (like TCAM width). For example, IPv6 Destination address (128b) and IPv4 Destination address (32b) keys can be fit using only 128 bits instead of 128 + 32bits.
+
```

Comment:
As mentioned in the "Restrictions/Limitations" section, IP\_TYPE or ETHER\_TYPE must be specified for ACL rules added in L3V4V6 ACL table to prevent a v4 packet matching an v6 rule and vice versa. This will also be used in phase-2 to decide on which underlying SAI ACL table a given ACL rule should be placed in.

Reply:
As mentioned in the "Restrictions/Limitations" section, IP\_TYPE or ETHER\_TYPE must be specified for ACL rules added in L3V4V6 ACL table to prevent a v4 packet matching an v6 rule and vice versa. This will also be used in phase-2 to decide on which underlying SAI ACL table a given ACL rule should be placed in.